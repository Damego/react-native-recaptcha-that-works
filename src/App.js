import React, {useRef, useCallback} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
  Button,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';
import Recaptcha from './Recaptcha';

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
  container: {
    backgroundColor: Colors.lighter,
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
});

const App = () => {
  const $recaptcha = useRef();

  const handlePless = useCallback(() => {
    $recaptcha.current.execute();
  }, []);
  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView style={styles.safeArea}>
        {/* <View
          contentInsetAdjustmentBehavior="automatic"
          style={styles.container}>
          <Button onPress={handlePless} title="Recaptcha" />
        </View> */}
        <Recaptcha
          ref={$recaptcha}
          siteKey="6LejsqwZAAAAAGsmSDWH5g09dOyNoGMcanBllKPF"
          onClose={() => console.log('onClose')}
          onError={() => console.log('onError')}
          onExpire={() => console.log('onExpire')}
          onVerify={(token) => console.log('onVerify', token)}
        />
      </SafeAreaView>
    </>
  );
};

export default App;
