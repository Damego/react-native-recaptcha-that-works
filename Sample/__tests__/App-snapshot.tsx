/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import App from '../App';

it('snapshot', () => {
  const {toJSON} = render(<App />);

  expect(toJSON()).toMatchSnapshot();
});
