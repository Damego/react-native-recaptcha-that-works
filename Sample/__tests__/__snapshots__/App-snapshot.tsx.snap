// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F3F3F3",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Open
        </Text>
      </View>
    </View>
    <Text>
      Token: 
      &lt;none&gt;
    </Text>
    <Text>
      Size: 
      invisible
    </Text>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
